app/src/main/java/info/nightscout/androidaps/dialogs/CarbsDialog.kt

```diff
@@ -270,10 +271,13 @@ class CarbsDialog : DialogFragmentWithDate() {
                     }
                     if (carbsAfterConstraints > 0) {
                         if (duration == 0) {
-                            uel.log("CARBS", d1 = carbsAfterConstraints.toDouble(), i1 = timeOffset)
+                            if (eventTimeChanged)
```

Comment:
would not be helpfull to do
`ValueWithUnit(condition = eventTimeChanged, eventTime, Units.Timestamp)` and add only if first param is true?
it's on so many places .....